### PR TITLE
Add ability to filter by card difficulty with list command (WIP - pending docs)

### LINKED ISSUE #270: Add ability to filter by performance
Related to #186

### PR DESCRIPTION
**Pull request by tomforge**
Resolves #270

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 1]
```diff
@@ -18,9 +23,32 @@
     public static final String COMMAND_WORD = "list";
 
     public static final String MESSAGE_SUCCESS_DECK = "Listed all decks";
-    public static final String MESSAGE_SUCCESS_CARD = "Listed all cards";
+    public static final String MESSAGE_SUCCESS_ALL_CARDS = "Listed all cards";
+    public static final String MESSAGE_SUCCESS_PERFORMANCE_CARDS = "Listed %1$s cards";
     public static final String AUTOCOMPLETE_TEXT = COMMAND_WORD;
```
Comment (leeyjjoel) [inline src/main/java/seedu/address/logic/commands/ListCommand.java]: You might want to update the autocomplete text to have a default difficulty

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns an ListCommand object for execution.
+     */
+    public ListCommand parse(String args) {
+        String trimmedArgs = args.trim();
+        if (!trimmedArgs.isEmpty()) {
+            String[] keywords = trimmedArgs.split("\\s+");
+            Set<Performance> performanceSet =
+                    Arrays.stream(keywords).filter(Performance::isValidPerformance)
```
Comment (leeyjjoel) [inline src/main/java/seedu/address/logic/parser/ListCommandParser.java]: I like this segment - the use of functional programming style makes the code elegant and clean

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 1 above)
Comment (leeyjjoel) [reply src/main/java/seedu/address/logic/commands/ListCommand.java]: Unless, of course, the preference is for the autocomplete to default to listing all the cards. In which case just leave it as it is

Comment (tomforge) [reply src/main/java/seedu/address/logic/commands/ListCommand.java]: I think it would be more intuitive to have the default case be listing all cards!

**Pull request merged**